service/matching/physical_task_queue_manager.go

```diff
@@ -348,9 +347,10 @@ func (c *physicalTaskQueueManagerImpl) PollTask(
 				SeriesName: pollMetadata.workerVersionCapabilities.DeploymentSeriesName,
 				BuildId:    pollMetadata.workerVersionCapabilities.BuildId,
 			}
-			d := deployment.NewDeploymentWorkflowClient(namespaceEntry, workerDeployment, c.partitionMgr.engine.historyClient)
-			err := d.RegisterTaskQueueWorker(ctx, c.queue.TaskQueueFamily().Name(), c.queue.TaskType(), nil, c.partitionMgr.engine.config.MaxIDLengthLimit())
 
+			err := c.partitionMgr.engine.deploymentStoreClient.RegisterTaskQueueWorker(
```

Comment:
nit: it's better to pass the deploymentStoreClient in the physical queues constructor rather than accessing the private fields of nested members. (Although we don't do a good job at this in the code currently)